internal/controllers/synthesis/lifecycle.go

```diff
@@ -112,12 +118,12 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 		// when the synthesized generation (from the status) changes, which will never happen because synthesis
 		// is righly disabled for deleted compositions. We break out of this deadlock condition by updating
 		// the status without actually synthesizing.
-		if comp.Status.CurrentSynthesis != nil && comp.Status.CurrentSynthesis.ObservedCompositionGeneration != comp.Generation {
+		if comp.Status.CurrentSynthesis != nil && (comp.Status.CurrentSynthesis.ObservedCompositionGeneration != comp.Generation || comp.Status.CurrentSynthesis.Synthesized == nil) {
```

Comment:
it's possible for the generations to match for reasons other than being set here, but we still need to enter this branch to set the synthesized timestamp.